File: scripts/ping_test.py
#!/usr/bin/python3
# -*- coding: utf-8 -*-


from dx2lib import *


def main():
    COMPort = '/dev/serial/by-id/usb-FTDI_FT232R_USB_UART_AI0282XE-if00-port0'
    Baudrate = 1000000

    dev = DX2_OpenPort(COMPort, Baudrate)
    if dev == None:
        print('Failed to open', COMPort)
        return
    err = c_uint16(0)

    DX2_SetTimeOutOffset(dev, 20)
    # ping
    print('>>DX2_Ping')
    for id in range(253):
        r = DX2_Ping(dev, id, err)
        print('\r ping = %d  err = %d     ' % (id, err.value), end='')
        if r:
            print('\n find', id)
            print(DXL_GetModelInfo(dev, id).contents.name.decode())
    print()
    DX2_ClosePort(dev)


if __name__ == '__main__':
    main()

File: src/ping_test.cpp
#include <stdio.h>
#include <string>
#include "dx2lib.h"

int main(int argc, char* argv[])
{
  if(argc < 2){
    printf("usage:%s com_port\n", argv[0]);
    exit(1);
  }
  TDeviceID dev;
  TErrorCode err;
  char *COMPort = argv[1];
  int Baudrate = 1000000;

  if ((dev = DX2_OpenPort(COMPort, Baudrate)))
  {
    printf("Open success\n");
    for (int i = 0; i <= 252; ++i)
    {
      if (DX2_Ping(dev, i, &err))
      {
        printf("\rFound     ID=%3d [$%04X]\n", i, err);
      }
      else
      {
        printf("\rNot found ID=%3d [$%04X]\r", i, err);
      }
    }

    DX2_ClosePort(dev);
  }
  else
  {
    printf("Failed to open %s\n", COMPort);
  }

  printf("\nFin\n");
}
